VILIB/Controllers/BookController.cs

```diff
@@ -33,10 +33,28 @@ public HttpResponseMessage Get()
             };
         }
 
-        //public HttpResponseMessage Put(string bookCode)
-        //{
-       
-        //}
+        public HttpResponseMessage Put(HttpPostedFileBase file)    //not tested
+        {
+            if (file.ContentLength > 0)
+            {
+                var fileName = Path.GetFileName(file.FileName);
+                var path = Path.Combine(HttpContext.Current.Server.MapPath("~/uploads"), fileName);
+                var decodedCode = _scannerPresenter.DecodedBarcode(path);
+                var book = _bookPresenter.FindBookByCode(decodedCode.Text);
+                var takenBook = _takenBookPresenter.AddTakenBook(book, StaticDataSource.CurrUser);
+                return new HttpResponseMessage
+                {
+                    Content = new StringContent(JsonConvert.SerializeObject(takenBook.HasToBeReturned),
```

Comment:
Is the manual serialization neccessary? cannot you define the serialize method in the Book class itself?

Reply:
I didn't think about that. I will fix this

Reply:
I didn't think about that. I will fix this